mods/furniture/init.lua

```diff
@@ -0,0 +1,100 @@
+furniture = {}
+
+furniture.parts = {
+	{
+		name = "chair",
+		description = "Chair",
+		mesh = "chair.obj",
+		collision_box = {
+			{-0.25, -0.50, -0.25, 0.25, 0.00, 0.25},
+			{-0.25, 0.00, 0.15, 0.25, 0.55, 0.25},
+		},
+	},
+	{
+		name = "big_chair",
+		description = "Big Chair",
+		mesh = "chair_big.obj",
+		collision_box = {
+			{-0.5, -0.5, -0.5, 0.5, 0.15, 0.5},
+			{-0.5, 0.15, 0.25, 0.5, 1.30, 0.5},
+		},
+		custom_tiles = {"wool_black.png"},
+	},
+	{
+		name = "flowerpot",
+		description = "Flower Pot",
+		mesh = "flowerpot.obj",
+		collision_box = {-0.4, -0.5, -0.4, 0.4, 0.5, 0.4},
+		custom_tiles = {"default_dirt.png",
+										"default_gold_block.png"},
+	},
+	{
+		name = "stool",
+		description = "Stool",
+		mesh = "stool.obj",
+		collision_box = {-0.35, -0.5, -0.25, 0.35, -0.1, 0.25},
+	},
+	{
+		name = "shelves",
+		description = "Shelves",
+		mesh = "shelves.obj",
+		collision_box = {0.5, 0.5, 0.5, -0.5, -0.5, -0.0625},
+	},
+}
+
+function furniture.register_for_base(base_node)
+	local base_definition = minetest.registered_nodes[base_node]
+
+	for _, part in ipairs(furniture.parts) do
+		-- We get an iterator function over substrings split by :
+		local base_node_full_name = string.gmatch(base_node, "([^:]+)")
```

Comment:
Same comment as raven's variations mod, overcomplicated way to do it, just cut to
`local base_node_name = string.match(base_node, ':(.*)')`